File: src/c/plb.c
#include <stdio.h>

int main() {
  char byte;
  size_t longest_bits = 0;
  size_t current_bits = 0;
  while ((byte = getchar()) != EOF) {
    for (int i = 0; i < 8; ++i) {
      if (0 == ((0b10000000 >> i) & byte)) {
        if (current_bits > longest_bits) {
          longest_bits = current_bits;
        }
        current_bits = 0;
      } else {
        ++current_bits;
      }
    }
  }
  if (current_bits > longest_bits) {
    longest_bits = current_bits;
  }
  printf("%zu\n", longest_bits);
  return 0;
}

File: src/java/plb.java
import java.io.IOException;

public class plb {
    public static void main(String[] args) throws IOException {
        int longestBits = 0;
        int currentBits = 0;
        int byteRead;
        while ((byteRead = System.in.read()) != -1) {
            for (int i = 0; i < 8; ++i) {
                if ((byteRead & (0b10000000 >> i)) == 0) {
                    if (currentBits > longestBits) {
                        longestBits = currentBits;
                    }
                    currentBits = 0;
                } else {
                    ++currentBits;
                }
            }
        }
        if (currentBits > longestBits) {
            longestBits = currentBits;
        }
        System.out.println(longestBits);
    }
}
